### PR TITLE
WIP Update Bevy to main

### PR DESCRIPTION
**Pull request by Vrixyz**
Currently testing on [bevy commit 9386bd0](https://github.com/bevyengine/bevy/commit/9386bd0114c44c9f00a2e9c41db1225aaa78d159)

- TODO: update to proper 0.15 and keep a lookout for https://github.com/bevyengine/bevy/pull/15969.

⚠️ big_space hasn't been updated.

Notable changes from bevy:
- orthographic scale has been removed: https://github.com/bevyengine/bevy/pull/15075
  - ⚠️ This is a difficult change to adapt to, the orthographic projection is broken (zooms too much?).
- orthographic projection default has 2 versions: 2d and 3d: https://github.com/bevyengine/bevy/pull/15073
- Msaa is now a component: https://github.com/bevyengine/bevy/pull/14273

### CODE REVIEW

File: src/controller/projections.rs [hunk 1]
```diff
@@ -90,7 +90,9 @@ pub fn update_orthographic(mut cameras: Query<(&mut EditorCam, &mut Projection,
         };
 
         let anchor_dist = editor_cam.last_anchor_depth().abs() as f32;
-        let target_dist = (editor_cam.orthographic.scale_to_near_clip * orthographic.scale).clamp(
+        let target_dist = (editor_cam.orthographic.scale_to_near_clip
+            * (orthographic.area.width() * orthographic.area.height()))
```
Comment (Vrixyz) [inline src/controller/projections.rs:94]: This is probably wrong, but I'm not sure what to put there.

File: src/extensions/dolly_zoom.rs [hunk 2]
```diff
@@ -222,8 +223,8 @@ impl DollyZoom {
                 *projection = proj_end.clone();
                 if let Projection::Orthographic(ortho) = &mut *projection {
                     let multiplier = ortho_tri_base_to_scale_factor(camera, ortho);
-
-                    ortho.scale = (*triangle_base * multiplier) as f32;
+                    let normalized_projection = ortho.scaling_mode / multiplier as f32;
+                    ortho.scaling_mode = normalized_projection * *triangle_base as f32;
```
Comment (Vrixyz) [inline src/extensions/dolly_zoom.rs:227]: modifications in this file are also difficult to assess for me.

File: src/controller/projections.rs (hunk 1 above)
Comment (aevyrie) [reply src/controller/projections.rs:94]: I think scale has been replaced with setting it here: https://dev-docs.bevyengine.org/bevy/render/camera/enum.ScalingMode.html

Comment (aevyrie) [reply src/controller/projections.rs:94]: It defaults to `ScalingMode::WindowSize(1.0)`, and now if you were using scale of 2.0, you would now do `ScalingMode::WindowSize(2.0)`. You will need to match on the scaling enum and extract the multiplier. I'm not sure if we can support anything other than windowsize, or if there is a better way to do this.

**Comment by Vrixyz** [comment]
- I'm closing this in favor of https://github.com/aevyrie/bevy_editor_cam/pull/27

**Comment by aevyrie** [comment]
Thanks for this effort, sorry I didn't find the time to finish this up.

**Comment by Vrixyz** [comment]
no problem, I'm glad we could help avoid ergonomics issues with the `scale` from projection!

**Pull request closed without merging**